lib/cstruct.ml

```diff
@@ -370,6 +370,35 @@ let of_bytes ?allocator buf =
     blit_from_bytes buf 0 c 0 buflen;
     set_len c buflen
 
+let string_fold ~f ~z str =
+  let st = ref z in
+  ( String.iter (fun c -> st := f !st c) str  ; !st )
+
+let of_hex str =
+  let hexdigit = function
+    | 'a' .. 'f' as x -> int_of_char x - 87
+    | 'A' .. 'F' as x -> int_of_char x - 55
+    | '0' .. '9' as x -> int_of_char x - 48
+    | x               -> Format.ksprintf invalid_arg "of_hex: `%c'" x
```

Comment:
May I suggest using `%C` here instead (escaping the argument),
alternatively `'%02x'" (int_of_char x)` ?
The latter will make it easier to spot the byte when trying to correlate with a `Cstruct.hexdump_pp`.

Reply:
done - keep in mind `i` is the index into the cstruct, I introduced `p` for position in the string, and output this on error